### PR TITLE
Write Value

### PR DESCRIPTION
**Pull request by dati91**
Implements the `write_value` method

### CODE REVIEW

**Comment by dati91** [comment]
@mmatyas @fokinv @zakorgy 
r?

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -486,14 +486,37 @@ impl BluetoothManager {
...
 
         sender.send(message).unwrap();
     }
 
-    pub fn write_value(&self, _id: String, _value: Vec<u8>, _sender: IpcSender<BluetoothObjectMsg>) {
-        unimplemented!()
+    pub fn write_value(&mut self, id: String, value: Vec<u8>, sender: IpcSender<BluetoothObjectMsg>) {
+        let mut adapter = match self.get_adapter() {
+            Some(a) => a,
+            None => send_error!(sender, "No adapter found"),
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:499]: `unwrap_or_else`?

File: components/script/dom/bluetoothremotegattdescriptor.rs [hunk 2]
```diff
@@ -113,4 +113,21 @@ impl BluetoothRemoteGATTDescriptorMethods for BluetoothRemoteGATTDescriptor {
             Ok(self.GetValue().unwrap())
         }
     }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattdescriptor-writevalue
+    fn WriteValue(&self, value: Vec<u8>) {
+        let (sender, receiver) = ipc::channel().unwrap();
+        self.get_bluetooth_thread().send(
+            BluetoothMethodMsg::WriteValue(self.get_instance_id(), value, sender)).unwrap();
+        let result = receiver.recv().unwrap();
```
Comment (mmatyas) [inline components/script/dom/bluetoothremotegattdescriptor.rs:122]: Does this always succeed?

File: components/net/bluetooth_thread.rs [hunk 3]
```diff
@@ -486,14 +486,37 @@ impl BluetoothManager {
...
+        };
+        let mut result = match self.get_gatt_characteristic(&mut adapter, &id) {
+            Some(c) => Some(c.write_value(value.clone())),
+            None => None,
+        };
+        if result.is_none() {
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:505]: `if let`

File: components/script/dom/bluetoothremotegattcharacteristic.rs [hunk 4]
```diff
@@ -150,4 +150,21 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
...
+        let (sender, receiver) = ipc::channel().unwrap();
+        self.get_bluetooth_thread().send(
+            BluetoothMethodMsg::WriteValue(self.get_instance_id(), value, sender)).unwrap();
+        let result = receiver.recv().unwrap();
+        match result {
+            BluetoothObjectMsg::BluetoothWriteValue => (),
+            BluetoothObjectMsg::Error {
+                error
+            } => {
+                println!("{}", error);
```
Comment (mmatyas) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:165]: error handling

File: components/script/dom/bluetoothremotegattcharacteristic.rs [hunk 5]
```diff
@@ -150,4 +150,21 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
             Ok(self.GetValue().unwrap())
         }
     }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-writevalue
+    fn WriteValue(&self, value: Vec<u8>) {
+        let (sender, receiver) = ipc::channel().unwrap();
+        self.get_bluetooth_thread().send(
+            BluetoothMethodMsg::WriteValue(self.get_instance_id(), value, sender)).unwrap();
```
Comment (zakorgy) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:158]: is this indentation ok? I mean

<pre>
 self.get_bluetooth_thread()
     .send( BluetoothMethodMsg::WriteValue(self.get_instance_id(), value, sender))
     .unwrap();
</pre>

would not be better?

Comment (dati91) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:158]: I used this for consistency, but there will be a style refactor patch where your solution will be used.

**Pull request merged**